src/device/CMakeLists.txt

```diff
@@ -4,6 +4,24 @@ project(DEVICE)
 # Check if python is installed with the necessary components
 find_package(Python3 QUIET REQUIRED COMPONENTS Interpreter)
 
+execute_process(
+    COMMAND ${Python3_EXECUTABLE} -m pip show jinja2
+    OUTPUT_QUIET
+    RESULT_VARIABLE EXIT_CODE
+)
+
+if(NOT ${EXIT_CODE} EQUAL 0)
+    message(
+        WARNING
+        "The Python3 package \"jinja2\" is not installed. We are going to install it."
+    )
+
+    execute_process(
+        COMMAND ${Python3_EXECUTABLE} -m pip install jinja2 --user
+        RESULT_VARIABLE EXIT_CODE
```

Comment:
I'm not good at cmake. is this RESULT_VARIABLE required? it looks like it is not used.